src/plugins/bfetch/public/streaming/from_streaming_xhr.ts

```diff
@@ -61,7 +61,11 @@ export const fromStreamingXhr = (
       if (signal) signal.removeEventListener('abort', onBatchAbort);
 
       if (isErrorStatus()) {
-        subject.error(new Error(`Batch request failed with status ${xhr.status}`));
+        const errorMsg =
+          xhr.status === 0
+            ? 'Network connection error'
```

Comment:
Looks fine to me, but in case you are looking for alternative options, how about:
`Error connecting to the Kibana server. Please check your network connection.`

Reply:
Do we know if the error only pertains to Kibana or could Kibana be passing on an ES error?

Reply:
How does this message appear in the UI? As just a description or with a title and description?

> Unable to connect to the Kibana server.  Please check your network connection and try again.

**With title**

> **Unable to connect to the Kibana server**
> Please check your network connection and try again.

Reply:
@gchaps I was updating the message for the toast seen here - https://github.com/elastic/kibana/issues/145810 - with your suggestion I'll look at changing the toast title as well.

Reply:
I've found some code with similar concerns and slightly different text -

Title: Cannot retrieve your data
Message: Check your network and proxy configuration. If the problem persists, contact your network administrator.

It seems to me that we should use the same text for both these cases.

Reply:
Agreed, we should use the same text.  

I prefer the second one.  Does that text also work for your situation?